src/utils/string.cc

```diff
@@ -135,13 +135,18 @@ std::string string_to_hex(const std::string& input) {
     return output;
 }
 
-
 std::string toHexIfNeeded(const std::string &str) {
+    return toHexIfNeeded(str, false);
+}
+
+std::string toHexIfNeeded(const std::string &str, bool escape_spec) {
+    // escape_spec: escape special chars or not
+    // spec chars: '"' (quotation mark, ascii 34), '\' (backslash, ascii 92)
     std::stringstream res;
 
     for (int i = 0; i < str.size(); i++) {
         int c = (unsigned char)str.at(i);
-        if (c < 32 || c > 126) {
+        if (c < 32 || c > 126 || (escape_spec == true && (c == 34 || c == 92))) {
```

Comment:
This is an ok alternative to what I suggested in the earlier review, but see comment regarding having two functions.